openstack/keystone/templates/etc/_secrets.conf.tpl

```diff
@@ -55,3 +55,10 @@ trusted_key_header = {{ .Values.api.cc_external.trusted_key_header | default "HT
 trusted_key_value = {{ .Values.api.cc_external.trusted_key_value }}
 {{- end }}
 {{- end }}
+
+{{- if .Values.report_invalid_password_hash.enabled }}
+[security_compliance]
+report_invalid_password_hash = "event"
+invalid_password_hash_secret_key = {{ required "invalid_password_hash_secret_key value is required when .Values.report_invalid_password_hash.enabled is true" .Values.report_invalid_password_hash.invalid_password_hash_secret_key | include "resolve_secret" }}
```

Comment:
Could you please explain why you have chosen `resolve_secret` instead of moving it all to a separate secrets config file?

Reply:
[`resolve_secret`](https://github.com/sapcc/helm-charts/blob/2cf71abfc62213dd9101c9d50d12a2543f36115e/openstack/utils/templates/_hosts.tpl#L23-L25) is here to interpolate Vault secret reference (provided in values files outside of this repository).

I've put the configuration into `_secrets.conf.tpl` as it fits nicely into `3)` of the current pattern of Keystone `oslo_config` configuration files, where there are: 1) general `keystone.conf` file with non-secret values; 2) logging-specific `logging.conf` file; 3) `keystone.conf.d/secrets.conf` file with sensitive values. I can't find any good reason to manage `report_invalid_password_hash` configuration in a separate file (separation of concerns might be one of them, but it's not relevant in this case)

Reply:
ah that's already secrets.conf. Sorry, i did not see it; my question makes no sense then.